src/app/game-state/character.service.ts

```diff
@@ -61,7 +63,11 @@ export class CharacterService {
         this.characterState.dead = true;
         this.characterState.reincarnate(); // make sure character reincarnation fires before other things reset
         this.reincarnationService.reincarnate();
-        this.characterState.dead = false;
+        // Revive the character in the next tick update for making sure that everything is stopped.
+        this.deathSubscriber = this.mainLoopService.tickSubject.subscribe(() => {
```

Comment:
So, I see death subscriber getting subscribed to, but I don't ever see a next() getting called for it. Seems to me this subject does nothing. Granted I'm looking at this on my phone, but what am I missing?